### PR TITLE
CE2: Add vendor-specific SY

### PR DESCRIPTION
**Pull request by ahmed-anas**
https://github.com/trilogy-group/ccab/issues/6540

debugging I printed the received cea in the diameter client and you can see that it now has the diameter sy application ID (the last item in the ven

image

### CODE REVIEW

File: diam/sm/sm_test.go [hunk 1]
```diff
@@ -98,6 +101,61 @@ func testStateMachine(t *testing.T, network string) {
...
+	}}
+	Supported-Vendor-Id {Code:265,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{10415}}
+	Vendor-Specific-Application-Id {Code:260,Flags:0x40,Length:32,VendorId:0,Value:Grouped{
+		Vendor-Id {Code:266,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{10415}},
+		Auth-Application-Id {Code:258,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{16777265}},
+	}}
+	Supported-Vendor-Id {Code:265,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{10415}}
+	Vendor-Specific-Application-Id {Code:260,Flags:0x40,Length:32,VendorId:0,Value:Grouped{
+		Vendor-Id {Code:266,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{10415}},
+		Auth-Application-Id {Code:258,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{16777302}},
```
Comment (ahmed-anas) [inline diam/sm/sm_test.go:146]: this is diameter Sy

File: go.mod [hunk 2]
```diff
@@ -10,3 +10,5 @@ require (
 	golang.org/x/net v0.0.0-20191007182048-72f939374954
 	google.golang.org/grpc v1.24.0
 )
+
+replace github.com/fiorix/go-diameter/v4 => ./
```
Comment (ahmed-anas) [inline go.mod:14]: without this, all internal imports are actually using the code from the original library (i.e. github.com/fiorix/go-diameter/v4). it's funny we survived so long without this.

File: diam/serve_test.go [hunk 3]
```diff
@@ -60,8 +60,8 @@ func TestCapabilitiesExchangeTLS(t *testing.T) {
 	srv.Config.ReadTimeout = tm
 	srv.Config.WriteTimeout = tm
 	srv.TLS = &tls.Config{
-		MinVersion: tls.VersionTLS10,
-		MaxVersion: tls.VersionTLS10,
+		MinVersion: tls.VersionTLS12,
```
Comment (ahmed-anas) [inline diam/serve_test.go:63]: tls 1.0 is very old and was not supported for me when running locally.

**Review by alaa-nassef — APPROVED** [review]
LGTM

**Review by alaa-nassef — APPROVED** [review]
LGTM

**Pull request merged**